site/app/templates/polls/AllPollsPageInstructor.twig

```diff
@@ -280,4 +282,69 @@
         });
     }
 
+    function exportPolls() {
+        let url = "{{base_url}}" + "/getPollExport";
+        $.ajax({
+            url: url,
+            type: "GET",
+            processData: false,
+            cache: false,
+            contentType: false,
+            error: function(err) {
+                console.error(err);
+                window.alert("Something went wrong. Please try again.");
+            },
+            success: function(response) {
+                let data = JSON.parse(response).data;
+                let text = JSON.stringify(data, null, "\t");
+                let element = document.createElement('a');
+                element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
+                element.setAttribute('download', "{{semester}}-{{course}}-polls.json");
+                element.style.display = 'none';
+                document.body.appendChild(element);
+                element.click();
+                document.body.removeChild(element);
+            }
+        });
+    }
+
+    function importPolls() {
```

Comment:
I'm confused at what this importPolls function is doing with ajax it looks like you're refreshing the page at the end of this anyways and only just uploading a file so it would be possible to just do this with a plain HTML form.

```html
<form action="{{base_url}}/importPolls">
  <input type="file" id="import-polls-area" name="polls-file">
  <input type="submit">
</form>
```

Reply:
The thing I don't like about this solution is that it makes you have to click a button for uploading the file, then click submit. The way I have it set up now makes it all one "Import" button instead of two separate buttons. I agree about the Ajax it was a kind of work around for something a little messy. I'll find a cleaner solution.

Reply:
You could add a `onChange` event on the file input that then fires a `submit()` event on the overall form, bypassing the need for hitting the submit button, but retain the overall "it's a form" simplicity @shailpatels is suggesting.